test/test_base.py

```diff
@@ -94,67 +111,53 @@ def start_server(self):
         self.server_thread = threading.Thread(
-            target=self.output_reader, args=(self.server.stdout, print)
+            target=self.output_reader, args=(self.server.stdout, self.log.debug)
         )
         self.server_thread.daemon = True
         self.server_thread.start()
 
-        # doesn't require anything container-related
-        print("\nWaiting for RPC")
+        self.log.info("Waiting for RPC")
         self.wait_for_rpc("scenarios_available")
 
-    # Quit
     def stop_server(self):
         self.cleanup()
 
     def wait_for_predicate(self, predicate, timeout=5 * 60, interval=5):
-        while True:
+        self.log.debug(f"Waiting for predicate with timeout {timeout}s and interval {interval}s")
+        while timeout > 0:
             if predicate():
-                break
+                return
             sleep(interval)
             timeout -= interval
-            if timeout < 0:
-                raise Exception("Timed out waiting for predicate Truth")
+        raise Exception("Timed out waiting for predicate Truth")
 
     def get_tank(self, index):
         wn = Warnet.from_network(self.network_name)
         return wn.tanks[index]
 
-    # Poll the warnet server for container status
-    # Block until all tanks are running
     def wait_for_all_tanks_status(self, target="running", timeout=20 * 60, interval=5):
+        """Poll the warnet server for container status
+        Block until all tanks are running
+        """
         def check_status():
             tanks = self.wait_for_rpc("network_status", {"network": self.network_name})
             stats = {"total": 0}
             for tank in tanks:
                 stats["total"] += 1
-                bitcoin_status = tank["bitcoin_status"]
-                if bitcoin_status not in stats:
-                    stats[bitcoin_status] = 0
-                stats[bitcoin_status] += 1
-                if "lightning_status" in tank:
-                    stats["total"] += 1
-                    lightning_status = tank["lightning_status"]
-                    if lightning_status not in stats:
-                        stats[lightning_status] = 0
-                    stats[lightning_status] += 1
-                if "circuitbreaker_status" in tank:
-                    stats["total"] += 1
-                    circuitbreaker_status = tank["circuitbreaker_status"]
-                    if circuitbreaker_status not in stats:
-                        stats[circuitbreaker_status] = 0
-                    stats[circuitbreaker_status] += 1
-            print(f"Waiting for all tanks to reach '{target}': {stats}")
-            # All tanks are running, proceed
+                for service in ["bitcoin", "lightning", "circuitbreaker"]:
+                    status = tank.get(f"{service}_status")
+                    if status:
+                        stats[status] = stats.get(status, 0) + 1
+            self.log.info(f"Waiting for all tanks to reach '{target}': {stats}")
```

Comment:
something else funky here, `total` shouldn't be the smallest number!

examples from local tests

```
 Waiting for all tanks to reach 'stopped': {'total': 4, 'stopped': 3, 'pending': 4, 'running': 2}

```





```
 Waiting for all tanks to reach 'stopped': {'total': 4, 'stopped': 9}
```

This test still stopped eventually because of
```
 ERROR - Error bringing network down: Timed out waiting for predicate Truth

```

Reply:
damn wth. I was so happy with that too. Is that happening every time? What test did you see that on?

It doesn't seem to for me:

![image](https://github.com/bitcoin-dev-project/warnet/assets/6606587/d8dcc066-6ace-43ed-8733-b532ec4424e4)


Will investigate more tests in a bit.

Reply:
try the ln_test because it has 4 tanks but a bunch of other pods for ln and cb

Reply:
Fixed in https://github.com/bitcoin-dev-project/warnet/commit/e1373a91bd7937de1d0dd5f7d746301e76b2c5d8